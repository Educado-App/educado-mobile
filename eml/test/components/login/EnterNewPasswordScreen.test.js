import renderer from 'react-test-renderer';
import React from 'react';
import EnterNewPasswordLayout from '../../../components/login/EnterNewPasswordScreen';

test('Enter new password screen renders correctly', async () => {
  let newPasswordLayout;
  await renderer.act(() => {
    newPasswordLayout = renderer.create(<EnterNewPasswordLayout />);
  });
  
  const tree = newPasswordLayout.toJSON();
  expect(tree).toMatchSnapshot();
});

test('toggleShowPassword toggles password visibility correctly', async () => {
  const newPasswordLayout = renderer.create(<EnterNewPasswordLayout />);
  const textField = newPasswordLayout.root.findByProps({ label: 'Nova senha' });
  const button = newPasswordLayout.root.findByProps({ id: 'showPasswordEye' });

  expect(textField.props.secureTextEntry).toBeTruthy();
  await renderer.act(() => {
    button.props.toggleShowPassword();
  });
  expect(textField.props.secureTextEntry).toBeFalsy();
});