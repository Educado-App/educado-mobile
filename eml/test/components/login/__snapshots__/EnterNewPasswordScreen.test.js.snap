// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enter new password screen renders correctly 1`] = `
<View>
  <View>
    <View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 8,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
              ],
            ]
          }
        >
          Nova senha
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 4,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#FF4949",
                },
              ],
            ]
          }
        >
          *
        </Text>
      </View>
      <View>
        <TextInput
          autoComplete="off"
          keyboardType="default"
          onChangeText={[Function]}
          passwordGuidelines={true}
          placeholder="Insira sua senha"
          secureTextEntry={true}
          style={
            [
              {
                "paddingBottom": 4,
                "paddingTop": 4,
              },
              {
                "paddingLeft": 10,
              },
              {
                "backgroundColor": "#FFFFFF",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
              {
                "borderBottomWidth": 2,
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
              },
              {
                "borderBottomColor": "#A1ACB2",
                "borderLeftColor": "#A1ACB2",
                "borderRightColor": "#A1ACB2",
                "borderTopColor": "#A1ACB2",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "position": "absolute",
          },
          {
            "right": 0,
          },
          {
            "top": 8,
          },
          {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "gray",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰈈
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "marginTop": 4,
        },
        {
          "height": 24,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#A1ACB2",
            },
          ],
        ]
      }
      testId="passwordLengthAlert"
    >
      • Mínimo 8 caracteres
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "transform": [
              {
                "translateY": -4,
              },
              {
                "translateX": 0,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "height": 24,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#A1ACB2",
            },
          ],
        ]
      }
      testId="passwordLetterAlert"
    >
      • Conter pelo menos uma letra
    </Text>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "transform": [
              {
                "translateY": -4,
              },
              {
                "translateX": 0,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#FF4949",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ],
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "marginTop": 24,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 8,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
              ],
            ]
          }
        >
          Confirmar nova senha
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "marginLeft": 4,
                },
                {
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#FF4949",
                },
              ],
            ]
          }
        >
          *
        </Text>
      </View>
      <View>
        <TextInput
          autoComplete="off"
          keyboardType="default"
          onChangeText={[Function]}
          passwordGuidelines={false}
          placeholder="Confirme sua senha"
          secureTextEntry={true}
          style={
            [
              {
                "paddingBottom": 4,
                "paddingTop": 4,
              },
              {
                "paddingLeft": 10,
              },
              {
                "backgroundColor": "#FFFFFF",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
              {
                "borderBottomWidth": 2,
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
              },
              {
                "borderBottomColor": "#A1ACB2",
                "borderLeftColor": "#A1ACB2",
                "borderRightColor": "#A1ACB2",
                "borderTopColor": "#A1ACB2",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "position": "absolute",
          },
          {
            "right": 0,
          },
          {
            "top": 8,
          },
          {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "gray",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰈈
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#FF4949",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ],
        ]
      }
    />
  </View>
  <View>
    <View
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#5ECCE9",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "opacity": 0.5,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "textAlign": "center",
              },
              {
                "fontFamily": "Montserrat-Bold",
              },
              {
                "fontSize": 16,
              },
              {
                "color": "#FFFFFF",
              },
            ],
          ]
        }
      >
        Entrar
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "center",
        },
        {
          "alignItems": "flex-end",
        },
        {
          "marginTop": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "color": "#A1ACB2",
            },
            {
              "lineHeight": 20,
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "marginRight": 4,
            },
          ],
        ]
      }
    >
      E-mail errado?
    </Text>
    <Text
      onPress={[Function]}
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "color": "#383838",
          },
          {
            "fontSize": 16,
          },
          [
            {
              "color": "#383838",
            },
            {
              "lineHeight": 20,
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "textDecorationLine": "underline",
            },
          ],
        ]
      }
    >
      Volte aqui
    </Text>
  </View>
</View>
`;
